src/conic_form.jl

```diff
@@ -0,0 +1,270 @@
+
+function sympackedUtoLidx(x::AbstractVector{<:Integer}, n)
+    y = similar(x)
+    map = sympackedLtoU(1:sympackedlen(n), n)
+    for i in eachindex(y)
+        y[i] = map[x[i]]
+    end
+    y
+end
+
+
+# Scale coefficients depending on rows index
+# rows: List of row indices
+# coef: List of corresponding coefficients
+# d: dimension of set
+# rev: if true, we unscale instead (e.g. divide by √2 instead of multiply for PSD cone)
+function _scalecoef(rows::AbstractVector{<: Integer}, coef::Vector{Float64}, d::Integer, rev::Bool)
+    scaling = rev ? 1 / √2 : 1 * √2
+    output = copy(coef)
+    for i in 1:length(output)
+        # See https://en.wikipedia.org/wiki/Triangular_number#Triangular_roots_and_tests_for_triangular_numbers
+        val = 8 * rows[i] + 1
+        is_diagonal_index = isqrt(val)^2 == val
+        if !is_diagonal_index
+            output[i] *= scaling
+        end
+    end
+    return output
+end
+
+# Unscale the coefficients in `coef` with respective rows in `rows` for a set `s`
+scalecoef(rows, coef, s) = _scalecoef(rows, coef, MOI.dimension(s), false)
+# Unscale the coefficients in `coef` with respective rows in `rows` for a set of type `S` with dimension `d`
+unscalecoef(rows, coef, d) = _scalecoef(rows, coef, sympackeddim(d), true)
+
+output_index(t::MOI.VectorAffineTerm) = t.output_index
+variable_index_value(t::MOI.ScalarAffineTerm) = t.variable_index.value
+variable_index_value(t::MOI.VectorAffineTerm) = variable_index_value(t.scalar_term)
+coefficient(t::MOI.ScalarAffineTerm) = t.coefficient
+coefficient(t::MOI.VectorAffineTerm) = coefficient(t.scalar_term)
+# constrrows: Recover the number of rows used by each constraint.
+# When, the set is available, simply use MOI.dimension
+constrrows(s::MOI.AbstractVectorSet) = 1:MOI.dimension(s)
+# When only the index is available, use the `conic.ncone.nrows` field
+constrrows(conic::ConicData, ci::CI{<:MOI.AbstractVectorFunction, <:MOI.AbstractVectorSet}) = 1:conic.cone.nrows[constroffset(conic, ci)]
+
+orderval(val, s) = val
+function orderval(val, s::MOI.PositiveSemidefiniteConeTriangle)
+    sympackedUtoL(val, s.side_dimension)
+end
+orderidx(idx, s) = idx
+function orderidx(idx, s::MOI.PositiveSemidefiniteConeTriangle)
+    sympackedUtoLidx(idx, s.side_dimension)
+end
+function __load_constraint(conic::ConicData, ci::MOI.ConstraintIndex, f::MOI.VectorAffineFunction, s::MOI.AbstractVectorSet)
+    func = MOIU.canonical(f)
+    I = Int[output_index(term) for term in func.terms]
+    J = Int[variable_index_value(term) for term in func.terms]
+    V = Float64[-coefficient(term) for term in func.terms]
+    offset = constroffset(conic, ci)
+    rows = constrrows(s)
+    conic.cone.nrows[offset] = length(rows)
+    i = offset .+ rows
+    b = f.constants
+    # @warn ci
+    # @warn offset
+    # @warn rows
+    if s isa MOI.PositiveSemidefiniteConeTriangle
+        b = scalecoef(rows, b, s)
```

Comment:
This is specific to SCS, as you can see in https://github.com/jump-dev/SCS.jl/pull/192/, the plan is to have solvers that need specific preprocessing for some constraints to use the `preprocess` keyword of `copy_to`.